src/First/README.md

```diff
@@ -0,0 +1,451 @@
+passed to the constructor will be wrapped in a `Just` to represent a non-empty
+instance of `First`. Any `Maybe` passed to the constructor will be lifted as
+is, allowing the ability to "choose" an value based on some disjunction.
+
+While most `Monoid`s only provide a [`valueOf`](#valueof) function used for
+extraction, `Last` takes advantage of its underlying `Maybe` to provide an
+additional [`option`](#option) method. Using [`valueOf`](#valueof) will extract
+the underlying `Maybe`, while [`option`](#option) will extract the underlying
+value in the `Maybe`, using the provided default value when the underlying
+`Maybe` is a `Nothing` instance.
+
+```javascript
+const First = require('crocks/First')
+
+const and = require('crocks/logic/and')
+const isNumber = require('crocks/predicates/isNumber')
+const mconcatMap = require('crocks/helpers/mconcatMap')
+const safe = require('crocks/Maybe/safe')
+
+// isEven :: Number -> Boolean
+const isEven =
+  x => !(x % 2)
+
+// isValid :: a -> Boolean
+const isValid =
+  and(isNumber, isEven)
+
+// chooseFirst :: [ * ] -> First Number
+const chooseFirst =
+  mconcatMap(First, safe(isValid))
+
+chooseFirst([ 21, 45, 2, 22, 19 ])
+  .valueOf(0)
+//=> Just 2
+
+chooseFirst([ 'a', 'b', 'c' ])
+  .option('')
+//=> ""
+```
+
+## Implements
+
+`Semigroup`, `Monoid`
+
+## Constructor Methods
+
+#### empty
+
+```haskell
+First.empty :: () -> First a
+```
+
+`empty` provides the identity for the `Monoid` in that when the value it
+provides is `concat`ed to any other value, it will return the other value. In
+the case of `First` the result of `empty` is `Nothing`. `empty` is available
+on both the Constructor and the Instance for convenience.
+
+```javascript
+const First = require('crocks/First')
+const { empty } = First
+
+First.empty()
+//=> First( Nothing )
+
+First(3)
+  .concat(empty())
+//=> First( Just 3 )
+
+empty()
+  .concat(First(3))
+//=> First( Just 3 )
+```
+
+## Instance Methods
+
+#### concat
+
+```haskell
+First a ~> First a -> First a
+```
+
+`concat` is used to combine (2) `Semigroup`s of the same type under an operation
+specified by the `Semigroup`. In the case of `First`, it will always provide the
+first non-empty value. Any subsequent non-empty values will be thrown away, and
+will always result in the first non-empty value.
+
+```javascript
+const First = require('crocks/First')
+const concat = require('crocks/pointfree/concat')
+
+const a = First('a')
+const b = First('b')
+const c = First('c')
+
+a.concat(b)
+//=> First( Just "a" )
+
+b.concat(a)
+//=> First( Just "b" )
+
+concat(c, concat(b, a))
+//=> First( Just "a" )
+
+concat(concat(c, b), a)
+//=> First( Just "a" )
+
+concat(concat(a, b), c)
+//=> First( Just "c" )
+```
+
+#### option
+
+```haskell
+First a ~> a -> a
+```
+
+`First` wraps an underlying `Maybe` which provides the ability to option out
+a value in the case as of an empty instance. Just like `option` on a `Maybe`
```

Comment:
is the `as` needed here? think it works fine as `...to option out a value in the case of an empty instance`